pkg/operator/ceph/csi/spec.go

```diff
@@ -408,6 +410,56 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	return nil
 }
 
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	// skip check if running unsupported images is allowed
+	if AllowUnsupported {
+		logger.Infof("Allowing unsupported ceph-csi images -> no version check")
+		return nil
+	}
+
+	ownerRef := metav1.OwnerReference{
```

Comment:
This will make the owner of the Job be the Job itself, which is redundant. It seems that the CSI driver code elsewhere is also doing the same thing, but I think it might be the wrong thing. In `StartCSIDrivers`, the CSI configmap is made an owner of itself, for example. Resources are generally given the owner as being the `CephCluster` resource. @travisn what are your thoughts?

Reply:
In any case we have to be very careful with owner references or else we run into resources disappearing as seen in #4590. For a temporary job like this I don't think it's critical to have the owner reference set, but I would vote to set it to the operator pod. If the operator pod is deleted/restarted, it's ok if this job goes away. We can't set it to the CephCluster CR since the csi driver is one-per-operator rather than one-per-cluster

Reply:
Changed the OwnerReference to the Rook-Operator.